src/site/content/en/blog/browser-nativefs/index.md

```diff
@@ -0,0 +1,366 @@
+the demo cannot be embedded in this article.
+
+## The browser-nativefs library in the wild
+
+In my free time, I contribute a tiny bit to an
+[installable PWA](https://web.dev/progressive-web-apps/#installable)
+called [Excalidraw](https://excalidraw.com/),
+a whiteboard tool that lets you easily sketch diagrams with a hand-drawn feel.
+It is fully responsive and works well from small mobile phones to big screen computers.
+This means it needs to deal with files on all the various platforms
+that may or, more likely, may not support the Native File System API,
+which makes it a great candidate for the browser-nativefs library.
+
+I can, for example, start a drawing on my iPhone,
+save it (technically: download it, since Safari does not support the Native File System API)
+to my iPhone Downloads folder, open the file on my desktop (after transferring it from my phone),
+modify the file, and over-save it with my changes, or even save it as a new file.
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="iphone-original.png" width="300" alt="An Excalidraw drawing on an iPhone.">
+  <figcaption class="w-figcaption">
+    Starting an Excalidraw drawing on an iPhone where the Native File System API is not supported,
+    but where a file can be saved (downloaded) to the Downloads folder.</a>)
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="chrome-modify.png" alt="The modified Excalidraw drawing on Chrome on the desktop.">
+  <figcaption class="w-figcaption">
+    Opening and modifying the Excalidraw drawing on the desktop where the Native File System API is supported
+    and thus the file can be accessed via the API.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="chrome-oversave.png" alt="Over-saving the original file with the modifications.">
+  <figcaption class="w-figcaption">
+    Over-saving the original file with the modifications to the original Excalidraw drawing file.
+    The browser shows a dialog asking me whether this is fine.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="chrome-save-as.png" alt="Saving the modifications to a new Excalidraw drawing file.">
+  <figcaption class="w-figcaption">
+    Saving the modifications to a new Excalidraw file. The original file remains untouched.
+  </figcaption>
+</figure>
+
+### Real life code sample
+
+Below, you can see an actual example of browser-nativefs as it is used in Excalidraw.
+The visible excerpt stems from the file
+[`/src/data/json.ts`](https://github.com/excalidraw/excalidraw/blob/cd87bd6901b47430a692a06a8928b0f732d77097/src/data/jsonts#L24-L52).
+Of special interest is how the `saveAsJSON()` method passes either a file handle or `null` to browser-nativefs'
+`fileSave()` method, which causes it to over-save when a handle is given,
+or to save to a new file if not.
+
+```js
+export const saveAsJSON = async (
```

Comment:
I don't see anything returned in this example either.

Reply:
Again good catch.